src/Integration.UnitTests/Suppression/ChecksumCalculatorTests.cs

```diff
@@ -0,0 +1,105 @@
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using SonarLint.VisualStudio.Integration.Suppression;
+
+namespace SonarLint.VisualStudio.Integration.UnitTests.Suppression
+{
+    [TestClass]
+    public class ChecksumCalculatorTests
+    {
+        [TestMethod]
+        public void Checksum_ThrowsOnNull()
+        {
+            Exceptions.Expect<ArgumentNullException>(() => ChecksumCalculator.Calculate(null));
+        }
+
+        [TestMethod]
+        public void Checksum_WhitespaceIsIgnored()
+        {
+            // 1. If the input only varies by whitespace then the checksums should be the same
+            var checksum1 = ChecksumCalculator.Calculate("abc");
+            var checksum2 = ChecksumCalculator.Calculate(" a  b  \r\n\t c \n\n");
+            Assert.AreEqual(checksum1, checksum2);
+
+            // 2. Logically, a whitespace-only string should have the same checksum as an empty tring
+            var emptyChecksum = ChecksumCalculator.Calculate("");
+            var whitespaceOnlyChecksum = ChecksumCalculator.Calculate("\r \t\n\n\r\t  ");
+            Assert.AreEqual(emptyChecksum, whitespaceOnlyChecksum);
+        }
+
+        [TestMethod]
+        public void Checksum_ExpectedValues()
+        {
+            // Alpha-numerical
+            AssertExpectedChecksum("1234567890abcdefghijklmnopqrstuvwxyz", "928f7bcdcd08869cc44c1bf24e7abec6");
+            AssertExpectedChecksum("1234567890ABCDEFGHIJKLMNOPQRSTUVWXYZ", "b8ea760cc575698b7aaf5afadd72f639");
+
+            // Punctuation
+            AssertExpectedChecksum("`-=¬!\"£$%^&*()_+)[];'#,./{}:@~<>?\\|", "3468fe90cd69365b03bbf015dd6374c3");
+
+            // Other special characters
+            AssertExpectedChecksum("\t\n\r\f\b\u0001\u058F\u319F\u23F0", "5378ea4677edcf84c17a91053a3fb0f9");
+            AssertExpectedChecksum("àÀáÁâÂèÈéÉêÊíÍòÒóÓôÔçÇùÙúÚûÛ", "55105c3496f8eb5ec78ab14304a5b96f");
+
+            // Empty string
+            AssertExpectedChecksum("", "d41d8cd98f00b204e9800998ecf8427e");
+        }
+
+        [TestMethod]
+        public void Checksum_RealIssueData()
+        {
+            // Real issues taken from an analysis of Akka.Net
+            AssertExpectedChecksum("        private HeliosTransport _transport;", "d60a97d7b258e0b4dfac90a65543eb50");
+
+            AssertExpectedChecksum("                Console.WriteLine(\"Performance benchmark starting...\");",
+                "7e42b5c3c797cb1e18c7d2d0ded681f5");
+
+            AssertExpectedChecksum("Console.WriteLine(\"All actors have been initialized...\");",
+                "a458ed466cea5d4a81253f622683a8f5");
+            
+            AssertExpectedChecksum("Console.WriteLine($\"{ActorCount} actors stored {MessagesPerActor} events each in {elapsed/1000.0} sec. Average: {ActorCount*MessagesPerActor*1000.0/elapsed} events/sec\"); ",
```

Comment:
Let's try to be careful with long lines (they are hard to read in GitHub). I set up a line length of 120 but you can get around 130 with GitHub.

Reply:
Ok in general; here, this line is a deliberately long real example with a number of special characters; it's not really meant to be read.

What do you mean by set up a line length of 120? Some configuration in the IDE?